src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -82,11 +83,42 @@ public boolean isDeleted() {
     }
 
     public List<Answer> getAnswers() {
-        return answers;
+        return answers.getAnswers();
+    }
+
+    public void validateDeletable(NsUser loginUser) throws CannotDeleteException {
+        validateOwner(loginUser);
+        answers.validateAnswerOwner(loginUser);
+    }
+
+    private void validateOwner(NsUser loginUser) throws CannotDeleteException {
+        if (!this.isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+    }
+
+    public List<DeleteHistory> delete(long questionId) {
```

Comment:
```suggestion
    public List<DeleteHistory> delete() {
```
questionid는 Question이 가지고 있음.
굳이 필요없지 않을까?